autorest/codegen/models/__init__.py

```diff
@@ -29,19 +32,22 @@
     "FileImport",
     "PrimitiveSchema",
     "LROOperation",
+    "OperationGroup",
     "PagingOperation",
+    "Property"
 ]
 
-# TODO: should this be in models.__init__ or CodeModel
-def build_schema(yaml_data, **kwargs):
-    code_model = kwargs.get('code_model')
+def build_schema(yaml_data: Dict[str, Any], **kwargs) -> BaseSchema:
+    code_model = kwargs.get('code_model', None)
+    if not code_model:
+        raise ValueError("CodeModel not passed through kwargs")
     yaml_id = id(yaml_data)
     try:
         return code_model.lookup_schema(yaml_id)
     except KeyError:
         # Not created yet, let's create it and add it to the index
         pass
-
+    schema: BaseSchema
```

Comment:
this lines does nothing, what were you trying to do?

Reply:
adding this line gets rid of a lot of mypy errors. If i don't add this, it assumes schema should be a ConstantSchema (bc that's the first one in the list of possible schema types), and the following are incompatible bc they are not ConstantSchemas or derive from ConstantSchema. If I add schema is a BaseSchema, since all of the possible schema types derive from BaseSchema there are not incompatible type warnings

Reply:
I think it's because there is no `= None` that I get confused, I didn't mypy would accept that as a type with no value.
Kind of scary